pkg/controller/job/job_controller.go

```diff
@@ -566,9 +567,39 @@ func (jm *Controller) syncJob(key string) (bool, error) {
 		}
 		if complete {
 			job.Status.Conditions = append(job.Status.Conditions, newCondition(batch.JobComplete, "", ""))
+			jobConditionsChanged = true
 			now := metav1.Now()
 			job.Status.CompletionTime = &now
 			jm.recorder.Event(&job, v1.EventTypeNormal, "Completed", "Job completed")
+		} else if *job.Spec.Suspend {
+			// Job can be in the suspended state only if it is NOT completed.
+			exists := false
+			for _, c := range job.Status.Conditions {
+				if c.Type == batch.JobSuspended {
+					exists = true
+					break
+				}
+			}
+			if !exists {
+				jm.recorder.Event(&job, v1.EventTypeNormal, "Suspended", "Job suspended")
+				job.Status.Conditions = append(job.Status.Conditions, newCondition(batch.JobSuspended, "", ""))
+				jobConditionsChanged = true
+			}
+		} else {
```

Comment:
we are in a a big nested if/else statement, if we don't want to refactor it in this PR, I suggest adding a comment beside each else. (e.g., in this case: `// Job not suspended`)